src/Tasks/sdk-tasks/GeneratePackagePruneDataDownloads.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Generates a list of targeting packs to download.  The package prune data will be extracted from these targeting packs
+    /// </summary>
+    public class GeneratePackagePruneDataDownloads : Task
+    {
+        [Required]
+        public string NETCoreAppTargetFrameworkVersion { get; set; }
+
+        [Output]
+        public ITaskItem[] TargetingPackDownloads { get; set; }
+
+        public override bool Execute()
+        {
+            List<TaskItem> targetingPackDownloads = new();
+
+            static TaskItem CreateDownload(string frameworkName, string netVersion)
+            {
+                var item = new TaskItem($"{frameworkName}.Ref");
+                item.SetMetadata("PackageVersion", $"{netVersion}.0");
+                item.SetMetadata("TargetFrameworkVersion", netVersion);
+                item.SetMetadata("FrameworkName", frameworkName);
+                return item;
+            }
+
+            void AddTargetingPackDownloads(string netVersion)
+            {
+                targetingPackDownloads.Add(CreateDownload("Microsoft.NETCore.App", netVersion));
+                targetingPackDownloads.Add(CreateDownload("Microsoft.AspNetCore.App", netVersion));
+                targetingPackDownloads.Add(CreateDownload("Microsoft.WindowsDesktop.App", netVersion));
+            }
+
+            int maxNetVersion = NuGetVersion.Parse(NETCoreAppTargetFrameworkVersion).Major;
+
+            //  Add targeting packs for .NET 10 and higher.  Targeting packs for previous versions don't have entirely accurate data,
+            //  so we use the FrameworkPackages data which was generated by running conflict resolution to determine which packages
+            //  would not be used.
+            //  HOWEVER: we don't download the targeting pack for the maximum .NET version here, as we may still be in preview.
+            //  Rather, the GetPackagesToPrune task will load the package prune data for the current version from the
+            //  targeting packs that ship with the SDK.
+            for (int netVersion = 10; netVersion < maxNetVersion; netVersion++)
```

Comment:
Isn't this defacto a no-op today with .NET 10 as `netVersion` is equal to `maxNetVersion`?

Reply:
OK you mentioned this in the PR description. Resolving.